### DISCUSSION TITLE
PBR Status in 4.0.0

### DISCUSSION THREAD

**Discussion by sava41** [raysan5/raylib#2493] [Q&A] [answered]
Hi,

I noticed the `models_material_pbr.c` example got removed in 4.0.0. I'm just getting started with with raylib and would be happy to port the example to 4.0.0 to better learn the API. Is it possible to get a rough overview of what needs to be done?

I got the example to compile on 4.0.0 with a few minor changes, but the helmet is flat shaded. The terminal shows a bunch of shader attributes are not found:
`WARNING: SHADER: [ID 7] Failed to find shader attribute: vertexTexCoord2`
`WARNING: SHADER: [ID 7] Failed to find shader attribute: vertexNormal`
`WARNING: SHADER: [ID 7] Failed to find shader attribute: vertexTangent`
`WARNING: SHADER: [ID 7] Failed to find shader attribute: vertexColor`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: mvp`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: matView`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: matProjection`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: matModel`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: matNormal`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: colDiffuse`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: texture0`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: texture1`
`WARNING: SHADER: [ID 7] Failed to find shader uniform: texture2`

Is the issue related to the shader attribute layout / naming conventions between versions 3.70 and 4.0.0 or is there something else?

Thanks,

Sava

**Comment by raysan5** [comment, marked as answer]
@sava41 Hi! The PBR example was mostly based on [LearnOpenGL PBR tutorial](https://learnopengl.com/PBR/Theory) and, actually, it was partially working but it required a full review of the implementation and the shaders and I removed it due to a lack of time to do that task. Another reson to remove was the amount of heavy assets it required.

You shouldn't have many problems to re-implement it for raylib 4.0 if you are up for the task.

The WARNING about shader attributes/uniforms not found shouldn't be a problem.
